kimchi/src/circuits/polynomials/keccak.rs

```diff
@@ -21,65 +77,265 @@ use crate::circuits::{
+//~ | Curr          | theta     | pirho        | chi           |
+//~
+//~ | `KeccakRound` | [0...100) |
+//~ | ------------- | --------- |
+//~ | Next          | iota      |
+//~
+//~ -----------------------------------------------------------------------------------------------------------------------------------------------------------------------
+//~
+//~ | Columns  | [0...100) | [100...120) | [120...200) | [200...220) | [220...240) | [240...260)  | [260...280) | [280...300)  | [300...320)  | [320...340) | [340...440) |
+//~ | -------- | --------- | ----------- | ----------- | ----------- | ----------- | ------------ | ----------- | ------------ | ------------ | ----------- | ----------- |
+//~ | theta    | state_a   | state_c     | shifts_c    | dense_c     | quotient_c  | remainder_c  | bound_c     | dense_rot_c  | expand_rot_c | state_d     | state_e     |
+//~
+//~ | Columns  | [440...840) | [840...940) | [940...1040) | [1040...1140) | [1140...1240) | [1240...1340) | [1340...1440) | [1440...1540) |
+//~ | -------- | ----------- | ----------- | ------------ | ------------- | ------------- | ------------- | ------------- | ------------- |
+//~ | pirho    | shifts_e    | dense_e     | quotient_e   | remainder_e   | bound_e       | dense_rot_e   | expand_rot_e  | state_b       |
+//~
+//~ | Columns  | [1540...1940) | [1940...2340) | [2340...2344 |
+//~ | -------- | ------------- | ------------- | ------------ |
+//~ | chi      | shifts_b      | shifts_sum    | f00          |
+//~
+//~ | Columns  | [0...4) | [4...100) |
+//~ | -------- | ------- | --------- |
+//~ | iota     | g00     | rest_g    |
+//~
+#[derive(Default)]
+pub struct KeccakRound<F>(PhantomData<F>);
+
+impl<F> Argument<F> for KeccakRound<F>
+where
+    F: PrimeField,
+{
+    const ARGUMENT_TYPE: ArgumentType = ArgumentType::Gate(GateType::KeccakRound);
+    const CONSTRAINTS: u32 = 754;
+
+    // Constraints for one round of the Keccak permutation function
+    fn constraint_checks<T: ExprOps<F>>(env: &ArgumentEnv<F, T>, _cache: &mut Cache) -> Vec<T> {
+        let mut constraints = vec![];
 
-    /// Creates Keccak rotation gates for the whole table (skipping the rotation by 0)
-    pub fn create_keccak_rot(
-        gates: &mut Vec<Self>,
-        new_row: usize,
-        zero_row: usize,
-    ) -> (usize, Vec<Self>) {
-        let mut rot_row = new_row;
-        for row in ROT_TAB {
-            for rot in row {
-                // if rotation by 0 bits, no need to create a gate for it
-                if rot == 0 {
-                    continue;
+        // DEFINE ROUND CONSTANT
+        let rc = [env.coeff(0), env.coeff(1), env.coeff(2), env.coeff(3)];
+
+        // LOAD STATES FROM WITNESS LAYOUT
+        // THETA
+        let state_a = grid!(100, env.witness_curr_chunk(0, 100));
+        let state_c = grid!(20, env.witness_curr_chunk(100, 120));
+        let shifts_c = grid!(80, env.witness_curr_chunk(120, 200));
```

Comment:
called `reset_c` in the rfc

Reply:
reset is the canonical representation, shifts is for the "masks" shifted. I thought that was updated in the RFC as well? Do you have pointers?

Reply:
https://github.com/o1-labs/rfcs/blob/dc2458ab708e9a380628ecf0e77fb30157cad7e0/0010-zkvm-keccak.md

Reply:
https://github.com/o1-labs/rfcs/blob/dc2458ab708e9a380628ecf0e77fb30157cad7e0/0010-zkvm-keccak.md#step-theta